pkg/alertmanager/amcfg.go

```diff
@@ -670,6 +682,30 @@ func (cb *configBuilder) convertWebhookConfig(ctx context.Context, in monitoring
 	return out, nil
 }
 
+func (cb *configBuilder) convertDiscordConfig(ctx context.Context, in monitoringv1alpha1.DiscordConfig, crKey types.NamespacedName) (*discordConfig, error) {
+	out := &discordConfig{
+		VSendResolved: in.SendResolved,
+		Title:         *in.Title,
+		Message:       *in.Message,
+	}
```

Comment:
you need to check for nil

```suggestion
	out := &discordConfig{
		VSendResolved: in.SendResolved,
	}
	
	if in.Title != nil && *in.Title != "" {
		out.Title = *in.Title
	}
	
	if in.Message != nil && *in.Message != "" {
		out.Message = *in.Message
	}
```

Reply:
@simonpasquier If I keep it `string` instead of `*string` then also it'll remain to be optional, right?

Reply:
Yes, but we want to keep them as pointers to be compliant with [kubernetes conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required).

With them as pointers, we need to check for nils 😬

Reply:
Thanks alot for the reference @ArthurSens